unitTests/ui/AnalyticsEndpointTest.ts

```diff
@@ -130,5 +130,36 @@ export function AnalyticsEndpointTest() {
         response: ['foo', 'bar', 'foobar']
       });
     });
+
+    it('should renew the token when the response is a 400 with the "InvalidToken" type', done => {
+      const newToken = 'new token';
+      spyOn(endpoint.options.accessToken, 'doRenew').and.callFake(() => {
+        endpoint.endpointCaller.options.accessToken = newToken;
+        return true;
+      });
+
+      spyOn(endpoint.endpointCaller, 'call').and.callFake(() => {
+        if (endpoint.endpointCaller.options.accessToken === newToken) {
+          return Promise.resolve({
+            statusCode: 200,
+            data: { searchEventResponses: [{ visitId: 'visitid' }] }
+          });
+        }
+
+        return Promise.reject({
+          statusCode: 400,
```

Comment:
It would be good to include a test where we return ten `400`s but assert that we only attempt to renew e.g. two times, to ensure we don't fall into a loop hitting the UA server.

Reply:
That part is in "doRenew" of the AccessToken class and is already tested